errata-email-notifications.py

```diff
@@ -0,0 +1,247 @@
+                content = CONTENT_TEMPLATE.substitute(
+                    errata_type=errata['type'].capitalize(),
+                    errata_severity=errata['severity'].capitalize(),
+                    errata_date=self.ts_to_date(errata['updated_date']['$date']),
+                    errata_description=errata['description'],
+                    errata_link=self.get_almalinux_errata_href(errata)
+                )
+
+                msg = EmailMessage()
+                msg['From'] = f'{EMAIL_FROM_NAME} <{self.sender}>'
+                msg['To'] = self.recipient
+                msg['Subject'] = subject
+                msg.set_content(content)
+
+                try:
+                    self.smtp_session.send(msg)
+                    self.save_last_processed_ts(last_processed_ts_file,
+                                                errata['updated_date']['$date'])
+                except:
+                    logging.error('Could not send errata notification email')
+
+        self.smtp_session.close()
+
+
+    def fetch_errata_data(self, url):
+        logging.debug('Fetching errata data from %s', url)
+        # TODO: If this script requires a considerable allocation of memory, we can
+        # figure out if we can publish the errata json files sorted by date.
+        # This way we can think about reading the response in chunks and stop reading
+        # when we reach the last_processed_ts. This could save some memory allocation
+        # during the execution of the script.
+        try:
+            response = request.urlopen(url)
+            data = json.loads(response.read())
+            # We want to return sorted erratas in descending order
+            sorted_data = sorted(data,
+                                 key=lambda x: x['updated_date']['$date'],
+                                 reverse=True)
+        except:
+            # Maybe more fine grained exception handling?
+            sorted_data = None
+        finally:
+            return sorted_data
+
+
+    def save_last_processed_ts(self, file_to_write, ts):
+        with open(os.path.join(BASEPATH, file_to_write), 'w') as f:
+            f.write(str(ts))
+        logging.info('Updated %s with timestamp %s', file_to_write, ts)
+
+
+    def ts_to_date(self, ts):
+        # We get ts in milliseconds
+        return datetime.utcfromtimestamp(ts/1000).strftime('%Y-%m-%d')
+
+
+    def get_almalinux_errata_href(self, errata):
+        href = ''
+        for ref in errata['references']:
+            # TODO: Better check if type equals to 'self'?
+            if ref['id'] == errata['updateinfo_id']:
+                href = ref['href']
+                break
+        return href
+
+
+if __name__ == '__main__':
+    args = parse_args()
+
+    # Set debug level
+    # TODO: Maybe allowing specific logging level?
+    log_level = logging.DEBUG if args.verbose else logging.WARNING
+    logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s', level=log_level)
```

Comment:
I'm think we need to add `logging.FileHandler` for this script to keep logs in some file

Reply:
Agree, will do, thanks!

Reply:
Ok, take a look at what I've done. When indicating -v or --verbose we won't log into a file but into console.
If not indicated, then the logs will go into a file, which can be defined by the user.
I've made it that way because I don't see the point of having logging statements go into both console and in the file, but let me know if you think otherwise.